scene/gui/view_panner.cpp

```diff
@@ -42,11 +42,11 @@ bool ViewPanner::gui_input(const Ref<InputEvent> &p_event, Rect2 p_canvas_rect)
 		// and one with pressed as false. Make sure we only process one of them.
 		if (scroll_vec != Vector2() && mb->is_pressed()) {
 			if (control_scheme == SCROLL_PANS) {
-				if (mb->is_ctrl_pressed()) {
+				if (mb->is_ctrl_pressed() && scroll_vec.y != 0) {
 					// Compute the zoom factor.
 					float zoom_factor = mb->get_factor() <= 0 ? 1.0 : mb->get_factor();
 					zoom_factor = ((scroll_zoom_factor - 1.0) * zoom_factor) + 1.0;
-					float zoom = (scroll_vec.x + scroll_vec.y) > 0 ? 1.0 / scroll_zoom_factor : scroll_zoom_factor;
+					float zoom = scroll_vec.y > 0 ? 1.0 / scroll_zoom_factor : scroll_zoom_factor;
 					zoom_callback.call(zoom, mb->get_position(), p_event);
 					return true;
 				} else {
```

Comment:
The `else` will activate when `scroll_vec.y == 0`, which can cause horizontal scrolling when holding Ctrl. I assume that's not desirable.

Reply:
Yes, in that version using SCROLL_PANS mode horizontal scroll pans with or wihtout Ctrl. But I agree, it would be more consistent not to pan with Ctrl. I changed that.